File: Algorithm/baekjoon/java/BJ10819.java
package baekjoon.java;

import java.util.*;

/**
 * Created by changmin on 2017. 12. 18..
 */
public class BJ10819 {
    public static boolean nextPermutation(int[] input){
        int i = input.length - 1;
        //뒤에서 부터 내림 차순이 안되는 i를 찾는다
        while(i > 0 && input[i-1] >= input[i]) i -= 1;

        //마지막인 경우
        if(i <= 0) return false;

        //오름 차순이 안되는 j를 찾아 i와 바꾼다
        int j = input.length - 1;
        while(input[j] <= input[i-1]) j -= 1;

        int temp = input[i-1];
        input[i-1] = input[j];
        input[j] = temp;

        //스왑한다.
        j = input.length - 1;
        while(i < j){
            temp = input[i];
            input[i] = input[j];
            input[j] = temp;
            i += 1; j -= 1;
        }

        return true;
    }

    public static int cal(int[] input){
        int output = 0;
        for(int i = 1; i < input.length; ++i){
            output += Math.abs(input[i] - input[i-1]);
        }
        return output;
    }

    public static int maxDifference(int[] input){
        int output = 0;
        Arrays.sort(input);
        do{
            int temp = cal(input);
            output = Math.max(temp, output);
        }while(nextPermutation(input));

        return output;
    }

    public static void main(String[] args){
        Scanner scanner= new Scanner(System.in);
        int size = scanner.nextInt();

        int[] inputs = new int[size];
        for(int i = 0; i < size; ++i){
           inputs[i] = scanner.nextInt();
        }

        int output = maxDifference(inputs);
        System.out.println(output);
    }
}

File: Algorithm/baekjoon/python/BJ10819.py
def next_permutation(array):
    i = len(array) - 1
    while i > 0 and array[i-1] >= array[i]:
        i -= 1
    
    if i <= 0:
        return False

    j = len(array) - 1
    while array[j] <= array[i-1]:
        j -= 1

    temp = array[i-1]
    array[i-1] = array[j]
    array[j] = temp

    j = len(array) - 1
    while i < j:
        temp = array[i]
        array[i] = array[j]
        array[j] = temp
        i += 1
        j -= 1

    return True

def max_difference(array):
    array.sort()

    output = get_value(array)

    while next_permutation(array):
        value = get_value(array)
        output = max(value, output)

    return output

def get_value(array):
    value = 0
    
    for i in range(1, len(array)):
        value += abs(array[i-1] - array[i])
    
    return value

def main():
    n = input()
    inputs = input().split(" ")
    array = list(map(int, inputs))
    print(max_difference(array))

main()